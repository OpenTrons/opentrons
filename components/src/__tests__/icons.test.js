// icon components tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {
  Icon,
  ALERT,
  BACK,
  CLOSE,
  REFRESH,
  SPINNER,
  USB,
  WIFI,
  FLASK,
  CHECKED,
  UNCHECKED,
  CHECKED_RADIO,
  UNCHECKED_RADIO,
  CHECKED_BOX,
  UNCHECKED_BOX,
  CHEVRON_UP,
  CHEVRON_DOWN,
  CHEVRON_LEFT,
  CHEVRON_RIGHT,
  FILE,
  COG,
  CONNECT,
  CONSOLIDATE,
  DISTRIBUTE,
  MIX,
  PAUSE,
  ARROW_RIGHT,
  MENU_DOWN
} from '..'

describe('icons', () => {
  test('ALERT icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={ALERT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('BACK icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={BACK} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CLOSE icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CLOSE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('REFRESH icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={REFRESH} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('SPINNER icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={SPINNER} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('spin prop adds spin class', () => {
    const icon = Renderer.create(
      <Icon name={SPINNER} spin />
    ).root.findByType('svg')

    expect(icon.props.className).toMatch(/\bspin\b/)
  })

  test('USB icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={USB} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('WIFI icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={WIFI} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('FLASK icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={FLASK} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHECKED icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHECKED} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('UNCHECKED icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={UNCHECKED} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHECKED_RADIO icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHECKED_RADIO} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('UNCHECKED_RADIO icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={UNCHECKED_RADIO} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHECKED_BOX icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHECKED_BOX} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('UNCHECKED_BOX icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={UNCHECKED_BOX} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHEVRON_UP icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHEVRON_UP} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHEVRON_DOWN icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHEVRON_DOWN} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHEVRON_LEFT icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHEVRON_LEFT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CHEVRON_RIGHT icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHEVRON_RIGHT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('FILE icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={FILE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('COG icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={COG} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CONNECT icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CONNECT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('ARROW_RIGHT icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={ARROW_RIGHT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('PAUSE icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={PAUSE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('MIX icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={MIX} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('DISTRIBUTE icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={DISTRIBUTE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('CONSOLIDATE icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CONSOLIDATE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('MENU_DOWN icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={MENU_DOWN} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
