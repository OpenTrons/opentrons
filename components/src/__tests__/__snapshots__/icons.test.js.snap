// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons alert icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M13.2 13.2h-2.4V6h2.4v7.2zm0 4.8h-2.4v-2.4h2.4V18zM12 0C5.373 0 0 5.373 0 12A12 12 0 1 0 12 0z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons back icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M18 11.242v1.516H8.91l4.166 4.166L12 18l-6-6 6-6 1.076 1.076-4.167 4.166z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons refresh icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 200 200"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M121.9,31.3L129,4.6l56.7,56.7l-77.5,20.8c0,0,7.7-28.6,7.7-28.6c-5.7-1.8-11.7-2.7-17.9-2.7 c-33.5,0-60.8,27.3-60.8,60.8c0,33.5,27.3,60.8,60.8,60.8c26.5,0,50.5-17.7,58.1-43.1l22,6.7c-5.1,16.8-15.7,32-29.8,42.6 c-14.6,11-32,16.8-50.3,16.8c-46.2,0-83.8-37.6-83.8-83.8S51.8,27.9,98,27.9C106.1,27.9,114.1,29.1,121.9,31.3"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons spinner icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 512 512"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons usb icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M15.347 6.792v4.529h1.132v2.264h-3.396V4.528h2.264L11.951 0 8.555 4.528h2.264v9.057H7.423v-2.343c.792-.42 1.358-1.223 1.358-2.185 0-1.37-1.12-2.491-2.49-2.491S3.8 7.686 3.8 9.056c0 .963.566 1.767 1.358 2.186v2.343a2.256 2.256 0 0 0 2.265 2.264h3.396v3.453a2.492 2.492 0 0 0-1.359 2.207 2.49 2.49 0 0 0 4.982 0c0-.962-.555-1.788-1.359-2.207v-3.453h3.396a2.256 2.256 0 0 0 2.264-2.264V11.32h1.132V6.792h-4.528z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons wifi icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M0 9.414l2.182 2.182c5.422-5.422 14.214-5.422 19.636 0L24 9.414c-6.622-6.622-17.367-6.622-24 0zm8.727 8.727L12 21.414l3.273-3.273a4.622 4.622 0 0 0-6.546 0zm-4.363-4.363l2.181 2.181a7.717 7.717 0 0 1 10.91 0l2.181-2.181c-4.21-4.211-11.05-4.211-15.272 0z"
    fillRule="evenodd"
  />
</svg>
`;
